### PR TITLE
feat(dfir_lang): Add flat_map_stream and flatten_stream operators to dfir_lang with tests

### PR DESCRIPTION
**Pull request by MingweiSamuel**
Created two new dfir_lang operators corresponding to dfir_pipes combinators:

- dfir_lang/src/graph/ops/flat_map_stream.rs: maps each item to a Stream
  via a closure and flattens the results, propagating Pending.
- dfir_lang/src/graph/ops/flatten_stream.rs: flattens items that are
  Streams, emitting their elements one by one, propagating Pending.

Both registered in declare_ops! macro in mod.rs.

Added dfir_rs/tests/surface_flat_stream.rs with 4 tests covering
both operators including empty stream edge cases.

### CODE REVIEW

**Comment by MingweiSamuel** [comment]
screwed up the jj somehow

File: hydro_lang/src/live_collections/stream/mod.rs [hunk 1]
```diff
@@ -618,6 +618,37 @@ where
...
+            self.location.clone(),
+            HydroNode::FlatMapStream {
+                f,
+                input: Box::new(self.ir_node.replace(HydroNode::Placeholder)),
+                metadata: self
+                    .location
+                    .new_node_metadata(Stream::<U, L, B, O, R>::collection_kind()),
+            },
+        )
+    }
```
[the remark this answers is not in the document]
Comment (MingweiSamuel) [reply hydro_lang/src/live_collections/stream/mod.rs]: fixed

**Review by shadaj — APPROVED** [review]
Let's create an issue to get rid of one of these later.
